sdk/core/System.ClientModel/src/Message/PipelineResponse.cs

```diff
@@ -94,23 +94,72 @@ internal bool TryGetBufferedContent(out MemoryStream bufferedContent)
         return false;
     }
 
-    internal void BufferContent(TimeSpan? timeout = default, CancellationTokenSource? cts = default)
+    internal void ProcessContent(bool bufferResponse, CancellationToken userToken, CancellationTokenSource joinedTokenSource)
```

Comment:
Since buffering is now owned by the transport, should this whole implementation be moved over to `PipelineTransport`?

Reply:
I think it makes sense to keep the buffering logic in the response, but I don't feel too strongly about this.  Only the part where we copy the content out of the network stream into a memory buffer is strictly needed in Response.  Do you see a strong advantage to moving the read-timeout-stream logic into the transport instead of holding it in the Response?

Reply:
I don't see a strong reason for it to live one place or the other except to clarify which component owns the concern. The transport still initiates it by calling `ProcessContent` on the response, but it seems more straight forward for it to handle the buffering logic itself directly. 

The reasons I'd say it makes more sense to stay in Response is if this should be invoked from other callers besides the transport or if we plan to implement additional response types where buffering may be implemented differently.

Reply:
BufferContent is invoked from ClientResultException, which is why that's needed in Response - I don't think we want to couple the transport into that functionality.  I can think a bit more about whether we need all of ProcessContent anywhere else, but it's also an implementation detail we can shift around after the fact, I think?

Reply:
Makes sense, thanks. 

One last question - why not retain the name `BufferContent` or some similar name that reflects the intent of "buffer or wrap in a read-only stream"

Reply:
> BufferContent is invoked from ClientResultException, which is why that's needed in Response

Aren't we moving towards a model evaluating `Content` "just works"? This would mean that only the transport would ever need to buffer explicitly.

Reply:
> Aren't we moving towards a model evaluating Content "just works"? This would mean that only the transport would ever need to buffer explicitly.

Sure, we can reevaluate what RequestFailedException/ClientResultException do when we close on what happens with response.Content.  I think until we close on that, we should preserve the behavior in the exception types where they buffer the response if that still needs to happen.

Reply:
I am going to add the response.Content change into this PR, so I'll close this thread since it reworks this area substantially.  If you have remaining comments, though, please add them in a new thread!